app/src/sharedTest/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagsFragmentTest.kt

```diff
@@ -0,0 +1,427 @@
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.SyncStatusModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.networking.NetworkConnectionDebugUtilModule
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [FeatureFlagsFragment]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = FeatureFlagsFragmentTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class FeatureFlagsFragmentTest {
+
+  @get:Rule val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
+  @get:Rule val oppiaTestRule = OppiaTestRule()
+  @Inject lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+  @Inject lateinit var platformParameterControllerDebugImpl: PlatformParameterControllerDebugImpl
+  @Inject lateinit var monitorFactory: DataProviderTestMonitor.Factory
+  @Inject lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+  }
+
+  @Test
+  fun testFeatureFlagsFragment_recyclerView_hasCorrectItemCount() {
+    launch(FeatureFlagsTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+
+      val expectedCount = getEphemeralFeatureFlags().size
+
+      onView(withId(R.id.feature_flags_recycler_view)).check { view, _ ->
+        val recyclerView = view as RecyclerView
+        assertThat(recyclerView.adapter?.itemCount).isEqualTo(expectedCount)
+        // Note to developers: if you add/remove a feature flag, please update the expected count.
+        assertThat(recyclerView.adapter?.itemCount).isEqualTo(14)
+      }
+    }
+  }
+
+  @Test
+  fun testFeatureFlagsFragment_recyclerViewItems_hasCorrectDetails() {
+    launch(FeatureFlagsTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+      getEphemeralFeatureFlags().forEachIndexed { index, ephemeralFeatureFlag ->
+        scrollToPosition(index)
+        verifyFeatureFlagDisplayName(
+          index,
+          getFeatureFlagDisplayName(ephemeralFeatureFlag.id)
+        )
+        verifyFeatureFlagSyncStatus(
+          index,
+          getSyncStatusText(ephemeralFeatureFlag.syncStatus)
+        )
+        verifyFeatureFlagSwitchState(
+          index,
+          ephemeralFeatureFlag.currentValue
+        )
+      }
+    }
+  }
+
+  @Test
+  fun testFeatureFlagFragment_downloadSupportFlag_hasCorrectDetails() {
```

Comment:
Here and elsewhere -- as it is `DOWNLOADS_SUPPORT`
```suggestion
  fun testFeatureFlagFragment_downloadsSupportFlag_hasCorrectDetails() {
```

Reply:
updated

Reply:
Please double-check other instances (local test variable names as well) 
https://github.com/oppia/oppia-android/blob/1c8a8798ce5686fb7d32d5932d5e1023c80b9c92/app/src/sharedTest/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagsFragmentTest.kt#L249
https://github.com/oppia/oppia-android/blob/1c8a8798ce5686fb7d32d5932d5e1023c80b9c92/app/src/sharedTest/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagsFragmentTest.kt#L272
https://github.com/oppia/oppia-android/blob/1c8a8798ce5686fb7d32d5932d5e1023c80b9c92/app/src/sharedTest/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagsFragmentTest.kt#L306

>You can do a quick search for "downloadSupport" to figure out if anything is missed.

Reply:
updated